library/ecp_curves_new.c

```diff
@@ -0,0 +1,5979 @@
+/*
+ *  Elliptic curves over GF(p): curve-specific data and functions
+ *
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *  not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "common.h"
+#include "ecp_invasive.h"
+
+#if defined(MBEDTLS_ECP_WITH_MPI_UINT)
```

Comment:
`MBEDTLS_ECP_WITH_MPI_STRUCT` and `MBEDTLS_ECP_WITH_MPI_UINT` should be mutually exclusive. 

As such we should be testing that:

    #if defined(MBEDTLS_ECP_WITH_MPI_UINT) &&  !defined MBEDTLS_ECP_WITH_MPI_STRUCT

Same should apply to ecp_curves.c